include/egg/core/eggFader.h

```diff
@@ -17,10 +17,10 @@ class Fader {
     virtual EStatus getStatus() const;
     virtual bool fadeIn();
     virtual bool fadeOut();
-    virtual BOOL calc();
+    virtual bool calc();
     virtual void draw();
 
-private:
+protected:
     EStatus mStatus;
```

Comment:
Should `mStatus` be a member of `ColorFader`? It definitely makes more sense being part of `Fader`, but I feel like if that were the case then `Fader::getStatus` wouldn't be a pure virtual function.

Reply:
Method bodies for `Fader` are not confirmed, aside from `Fader::Fader() {}`. We don't know if the virtual functions are pure virtual or not.

I am currently leaning towards `mStatus` being a member of `Fader`. Consider the virtual function `fadeIn`, and its definition for `ColorFader`:

```cpp
bool ColorFader::fadeIn() {
    bool success = mStatus == FADER_STATUS_OPAQUE;
    if (success) {
        mStatus = FADER_STATUS_FADE_IN;
        mFrameTimer = 0;
    }

    return success;
}
```

It's possible for `Fader`'s definition to be this:

```cpp
bool Fader::fadeIn() {
    bool success = mStatus == FADER_STATUS_OPAQUE;
    if (success) {
        mStatus = FADER_STATUS_FADE_IN;
    }

    return success;
}
```

Until we have ironed out more details from other compiled binaries, this is the assumption I'm operating under.

Reply:
Might be worth looking into sead to see if we can get any more information. I know the concepts originate from JSystem's JUTFader (for color only), which was then split into Fader (interface?) and ColorFader. Perhaps Fader lives on in the Switch era.